src/rawchannel.cc

```diff
@@ -122,9 +130,18 @@ class RawChannel : public Nan::ObjectWrap
       if (ioctl(m_SocketFd, SIOCGIFINDEX, &ifr) != 0)
         goto on_error;
 
+      // Configuration updated to use the CAN_FD 
       err_mask = CAN_ERR_MASK;
-      if (setsockopt(m_SocketFd, SOL_CAN_RAW, CAN_RAW_ERR_FILTER, &err_mask, sizeof(err_mask)) != 0)
-        goto on_error;
+      Flag_CANFD_Used = 1 ;                                                                               // GT MODIF : try to use CAN_FD first
```

Comment:
you shouldn't do this global. each time a new channel is constructed the global variable is touched, so using mixed CAN HS and FD channels will fail. Use a member variable instead.

Reply:
Sorry I don't know how to do that.
**Could you help me.**
I'm not expert in java and I'm confuse about the Class & member.